Mathlib/Topology/Compactness/Compact.lean

```diff
@@ -539,14 +539,10 @@ omit [TopologicalSpace X] in
 theorem isCompact_generateFrom' [T : TopologicalSpace X]
     {S : Set (Set X)} (hTS : T = generateFrom S) {s : Set X}
     (h : ∀ (ι : Type u) (U : ι → S), s ⊆ ⋃ i, U i → ∃ J : Set ι, J.Finite ∧ s ⊆ ⋃ i ∈ J, U i) :
-    IsCompact s := by
-  apply isCompact_generateFrom hTS
-  intro P hP hs
-  rw [Set.sUnion_eq_iUnion] at hs
-  obtain ⟨J, hJ, cover⟩ := h P (fun a ↦ ⟨a.1, hP a.2⟩) hs
-  refine ⟨(fun x ↦ x.1) '' J, ⟨by simp, Set.Finite.image _ hJ, ?_⟩⟩
-  simpa only [Set.sUnion_eq_iUnion, Set.iUnion_coe_set, Set.mem_image, Subtype.exists,
-    exists_and_right, exists_eq_right, Set.iUnion_exists] using cover
+    IsCompact s :=
+  isCompact_generateFrom hTS fun P hP hs ↦
+    have ⟨J, hJ, cover⟩ := h P (fun a ↦ ⟨a.1, hP a.2⟩) (sUnion_eq_iUnion ▸ hs)
+    ⟨(·.1) '' J, ⟨by simp, hJ.image _, by aesop⟩⟩
```

Comment:
@alreadydone This one, and the one using dot notation below, I think are the only ones of these that have *mild* value. The rest is just churn. That is, the rest of this PR is what I consider the "bad" form of golfing (i.e., shortening a proof for the sake of shortening it) as opposed to the "good" form of golfing, where a proof is shorter or more readable as a consequence of using more automation (without significant performance hit) or restructuring the proof to use a better technique. In the future, I suggest keeping only golfs that adhere to the "good" form. I going to merge this as-is because the golfs I don't think are a net negative, and I don't want to have anyone spend more time on this than you reading this comment.

Reply:
I mostly agree. I came across the PR adding this lemma and wondered why it's needed, so I looked into the proof and found indeed there are some nontrivial transformations necessary to deduce this form, and also noticed some simplifications. Then I looked into related proofs and golfed them in passing. I'm not working on something related so I can't bundle this in some PR, so I opened a standalone PR. It's not my intention to waste maintainer time but it doesn't seem ideal to shelve and forget about it either. I left a comment at the original PR #32213 hoping the original reviewer would take a quick look and merge this quickly so as not to waste additional reviewer time, but unfortunately it didn't happen.